A66-otel-stats.md

```diff
@@ -0,0 +1,542 @@
+with OpenTelemetry suggested as the successor framework.
+
+### Related Proposals:
+
+*   [A6: gRPC Retry Design](A6-client-retries.md)
+*   [A39: xDS HTTP Filter Support](A39-xds-http-filters.md)
+*   [A45: Exposing OpenCensus Metrics and Tracing for gRPC retry](A45-retry-stats.md)
+
+## Proposal
+
+### Metrics Schema
+
+#### Units
+
+Following the
+[OpenTelemetry Metrics Semantic Conventions](https://opentelemetry.io/docs/specs/otel/metrics/semantic_conventions/),
+the following units are used -
+
+*   Latencies are measured in float64 seconds, `s`
+*   Sizes are measured in bytes, `By`
+*   Counts for number of calls are measured in `{call}`
+*   Counts for number of attempts are measured in `{attempt}`
+
+Buckets for histograms in default views should be as follows -
+
+*   Latency : 0, 0.00001, 0.00005, 0.0001, 0.0003, 0.0006, 0.0008, 0.001, 0.002,
+    0.003, 0.004, 0.005, 0.006, 0.008, 0.01, 0.013, 0.016, 0.02, 0.025, 0.03,
+    0.04, 0.05, 0.065, 0.08, 0.1, 0.13, 0.16, 0.2, 0.25, 0.3, 0.4, 0.5, 0.65,
+    0.8, 1, 2, 5, 10, 20, 50, 100
+*   Size : 0, 1024, 2048, 4096, 16384, 65536, 262144, 1048576, 4194304,
+    16777216, 67108864, 268435456, 1073741824, 4294967296
+*   Count : 0, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192,
+    16384, 32768, 65536
+
+These buckets were chosen to maintain compatibility with the gRPC OpenCensus
+spec. The OpenTelemetry API has added an experimental feature for
+[advice](https://opentelemetry.io/docs/specs/otel/metrics/api/#instrument-advice)
+that would allow the gRPC library to provide these buckets as a hint. Since this
+is still an experimental feature and not yet implemented in all languages, it is
+upto the user to choose the right bucket boundaries.
+
+Also note that, as per an
+[OpenTelemetry proposal on stability](https://docs.google.com/document/d/1Nvcf1wio7nDUVcrXxVUN_f8MNmcs0OzVAZLvlth1lYY/edit#heading=h.dy1cg9doaq26)
+though, changes to bucket boundaries might not be considered a breaking change.
+Depending on the proposal, this recommendation would change to use
+`ExponentialHistogram`s instead, which would allow for automatic adjustments of
+the scale to better fit the data.
+
+#### Attributes
+
+*   `grpc.method` : Full gRPC method name, including package, service and
+    method, e.g. "google.bigtable.v2.Bigtable/CheckAndMutateRow". Note that some
+    gRPC implementations allow server to handle generic method names, i.e., not
+    registering method names in advance with the server. This allows clients to
+    send arbitrary method names that could potentially open up the server to
+    malicious attacks that result in metrics being stored with a high
+    cardinality. To prevent this, unregistered/generic method names should by
+    default be reported with "other" value instead. Implementations should
+    provide the option to override this behavior to allow recording generic
+    method names as well.
```

Comment:
This talks about generic methods, but does mention arbitrary methods from a headers frame (unrelated to if the server supports generic methods). I guess technically this is semantically covered by the statement "allows clients to send arbitrary method names", but I don't like the language, since it's just arbitrary headers frames from the wire, not whether the client is allowed to or not.

Reply:
Discussed offline and updated wording